BlueDB/src/main/java/io/bluedb/api/BlueQuery.java

```diff
@@ -16,7 +15,7 @@
 	BlueQuery<T> afterOrAtTime(long time);
 
 	List<T> getList() throws BlueDbException;
-	Iterator<T> getIterator() throws BlueDbException;
+	CloseableIterator<T> getIterator() throws BlueDbException;

```

Comment:
I would prefer returning a class with stream in the name so that it screams "This is a resource and must be used in a try with resource!". Iterator doesn't scream that at me. Either that or make the iterator name contain resource or something. Then depending on what the class is we could change the method name to match.